### PR TITLE
#199988 user dashboard

### PR DESCRIPTION
**Pull request by snehalsabade**
Please review the PR for user-dashboard.
PR: https://github.com/tekdi/sunbird-admin-ui/pull/20
- create user dashboard
- show user role and user role count in table
- user can sort and search by user role

Implementation video: https://www.loom.com/share/211527fc4212469592dd9836a2925861?sid=e569d616-4322-4801-b1e1-b3e3903d991a

### CODE REVIEW

File: src/app/sb-admin/components/pages/sb-dashboard/sb-dashboard.component.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<div class="grid">
+    <div class="col-12 xl:col-12">
+        <div class="card">
+            <h5>{{ 'USER_DASHBOARD' | i18next }}</h5>
+            <p-message severity="warn" text="{{ 'USER_ROLE_DASHBOARD_LIST_WARNING' | i18next }}" 
+            *ngIf="!userRoleCount || userRoleCount.length === 0"></p-message>
+            <p-table #myTable [value]="userRoleCount" [loading]="loading" [paginator]="true" [rows]="10"
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-dashboard/sb-dashboard.component.html]: myTable => sbDashboardTable

**Review by gouravmore — APPROVED** [review]

File: src/app/sb-admin/components/pages/sb-dashboard/sb-dashboard.component.html (hunk 1 above)
Comment (snehalsabade) [reply src/app/sb-admin/components/pages/sb-dashboard/sb-dashboard.component.html]: Resolved

**Review by gouravmore — APPROVED** [review]

File: src/locales/en/en.json [hunk 2]
```diff
@@ -69,6 +69,12 @@
   "USER_EDIT_ROLE":"Edit User Role",
   "USER_ROLES":"Roles",
   "USER_ROLE_ADDED":"User Role Updated Successfully",
+
+  "USER_DASHBOARD_ROLE_NAME":"User Role",
```
Comment (manojLondhe) [inline src/locales/en/en.json:75]: "USER_DASHBOARD_ROLE_NAME": "User Role",
"USER_DASHBOARD_USERS_ROLE_COUNT": "Count",
"USER_ROLE_DASHBOARD_LIST_WARNING": "Users Data Not Found",
"USER_DASHBOARD": "Dashboard",
"USER_DASHBOARD_ROLE": "Enter User Role",

File: src/config/url.config.json [hunk 3]
```diff
@@ -2,10 +2,12 @@
     "URLS": {
         "ORGANIZATION_CREATE_URL": "api/org/v1/create",
         "ORGNIZATION_SEARCH_URL": "api/org/v1/search",
-        "USER_BLOCK_URL": "api/user/v1/block",
         "USER_CREATE_URL": "api/user/v3/create",
         "USER_UPDATE_ROLE_URL": "api/user/v1/role/assign",
+        "USER_BLOCK_URL": "api/user/v1/block",
         "USER_UNBLOCK_URL": "api/user/v1/unblock",
+        "USER_DASHBOARD_ALL_USER_ROLE_URL":"api/data/v1/form/read",
+        "USER_DASHBOARD_USERS_ROLE_COUNT":"api/user/v1/search",
```
Comment (manojLondhe) [inline src/config/url.config.json]: use USER_SEARCH_URL instead of USER_DASHBOARD_USERS_ROLE_COUNT

File: src/config/url.config.json [hunk 4]
```diff
@@ -2,10 +2,12 @@
     "URLS": {
         "ORGANIZATION_CREATE_URL": "api/org/v1/create",
         "ORGNIZATION_SEARCH_URL": "api/org/v1/search",
-        "USER_BLOCK_URL": "api/user/v1/block",
         "USER_CREATE_URL": "api/user/v3/create",
         "USER_UPDATE_ROLE_URL": "api/user/v1/role/assign",
+        "USER_BLOCK_URL": "api/user/v1/block",
         "USER_UNBLOCK_URL": "api/user/v1/unblock",
+        "USER_DASHBOARD_ALL_USER_ROLE_URL":"api/data/v1/form/read",
```
Comment (manojLondhe) [inline src/config/url.config.json]: USER_DASHBOARD_ALL_USER_ROLE_URL => USER_ROLES_GET_URL

File: src/app/layout/app.menu.component.ts [hunk 5]
```diff
@@ -35,6 +35,11 @@ export class AppMenuComponent implements OnInit {
                 label: 'User Management',
                 icon: 'pi pi-fw pi-briefcase',
                 items: [
+                    {
+                        label: 'User Dashboard',
```
Comment (manojLondhe) [inline src/app/layout/app.menu.component.ts]: can labels be translated?

File: src/app/sb-admin/api/user.ts [hunk 6]
```diff
@@ -19,6 +19,10 @@ export interface User {
     organisations:any;
 }
 
+export interface userRoleCountModel {
```
Comment (manojLondhe) [inline src/app/sb-admin/api/user.ts:22]: format code pls

File: src/app/sb-admin/service/user.service.ts [hunk 7]
```diff
@@ -54,4 +54,22 @@ export class UserService {
       return this.http.post(config.URLS.USER_UNBLOCK_URL, payload, { headers: header });
     }
   }
-}
\ No newline at end of file
+
+  getAllUserRole(body: any): Observable<Object> {
```
Comment (manojLondhe) [inline src/app/sb-admin/service/user.service.ts]: getAllUserRole => getAllUserRoles

File: src/app/sb-admin/service/user.service.ts [hunk 8]
```diff
@@ -54,4 +54,22 @@ export class UserService {
...
+    let header = new HttpHeaders({
+      "Content-Type": 'application/json',
+      "Authorization": environment.authKey,
+      "x-authenticated-user-token": environment.userToken,
+    })
+    return this.http.post(config.URLS.USER_DASHBOARD_ALL_USER_ROLE_URL, body, { headers: header })
+  }
+
+  getUserRoleCount(body: any): Observable<Object> {
```
Comment (manojLondhe) [inline src/app/sb-admin/service/user.service.ts]: getUserRoleCount => getUsersCountByRole

File: src/app/sb-admin/components/pages/sb-dashboard/sb-dashboard.component.html [hunk 9]
```diff
@@ -0,0 +1,32 @@
+<div class="grid">
```
Comment (manojLondhe) [inline src/app/sb-admin/components/pages/sb-dashboard/sb-dashboard.component.html:1]: rename new page inside /components/pages/sb-user
name new page as user-dashboard

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/api/user.ts (hunk 6 above)
Comment (snehalsabade) [reply src/app/sb-admin/api/user.ts:22]: Done

File: src/app/sb-admin/components/pages/sb-dashboard/sb-dashboard.component.html (hunk 9 above)
Comment (snehalsabade) [reply src/app/sb-admin/components/pages/sb-dashboard/sb-dashboard.component.html:1]: done

File: src/app/sb-admin/service/user.service.ts (hunk 7 above)
Comment (snehalsabade) [reply src/app/sb-admin/service/user.service.ts]: Done sir

File: src/app/sb-admin/service/user.service.ts (hunk 8 above)
Comment (snehalsabade) [reply src/app/sb-admin/service/user.service.ts]: Done

File: src/config/url.config.json (hunk 4 above)
Comment (snehalsabade) [reply src/config/url.config.json]: Done

File: src/locales/en/en.json (hunk 2 above)
Comment (snehalsabade) [reply src/locales/en/en.json:75]: Done sir

File: src/config/url.config.json (hunk 3 above)
Comment (snehalsabade) [reply src/config/url.config.json]: Done

File: src/app/layout/app.menu.component.ts (hunk 5 above)
Comment (snehalsabade) [reply src/app/layout/app.menu.component.ts]: Done sir

**Pull request merged**